src/VisualStudio/VisualBasic/Impl/Progression/VisualBasicProgressionLanguageService.vb

```diff
@@ -14,39 +14,31 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.Progression
     Partial Friend Class VisualBasicProgressionLanguageService
         Implements IProgressionLanguageService
 
-        Public Function GetTopLevelNodesFromDocument(root As SyntaxNode, cancellationToken As CancellationToken) As IEnumerable(Of SyntaxNode) Implements IProgressionLanguageService.GetTopLevelNodesFromDocument
-            ' TODO: Implement this lazily like in C#?
+        Public Iterator Function GetTopLevelNodesFromDocument(root As SyntaxNode, cancellationToken As CancellationToken) As IEnumerable(Of SyntaxNode) Implements IProgressionLanguageService.GetTopLevelNodesFromDocument
+            If cancellationToken.IsCancellationRequested Then Exit Function
```

Comment:
❗️ This is a problem. It requires that the caller always check the cancellation at the end of iteration to ensure it's not cancelled. The return value of a method needs to reveal the validity of the result in cases where cancellation does not produce an exception.

The easiest fix is restoring the original `ThrowIfCancellationRequested` behavior.

Reply:
I'm following the [C# Implementation](https://github.com/dotnet/roslyn/blob/6847f1e5a909395aae9456e8f366cbf4deb86b69/src/VisualStudio/CSharp/Impl/Progression/CSharpProgressionLanguageService.cs#L46), with the check un-negated to enable early exiting.

Reply:
do we use "then exit function" anywhere else in the IDE?  I'd prefer us not use random non-common styles like this.

Reply:
@CyrusNajmabadi [We do](https://github.com/dotnet/roslyn/blob/160c4098fa90cff36fadad616eab2ae967ad2563/src/VisualStudio/VisualBasic/Impl/CodeModel/VisualBasicCodeModelService.vb#L492). There maybe use of the form `... Then Return`, but searching the codebase for that example is tricky. *For me, `Exit Function` makes more sense, as we're not "returning" anything.